pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -224,38 +217,90 @@ func (rm *ResourceQuotaController) syncResourceQuota(quota api.ResourceQuota) (e
-func PodCPU(pod *api.Pod) *resource.Quantity {
-	val := int64(0)
-	for j := range pod.Spec.Containers {
-		val = val + pod.Spec.Containers[j].Resources.Limits.Cpu().MilliValue()
+// PodsRequests returns sum of each resource request for each pod in list
+// If a given pod in the list does not have a request for the named resource, we log the error
+// but still attempt to get the most representative count
+func PodsRequests(pods []*api.Pod, resourceName api.ResourceName) *resource.Quantity {
+	var sum *resource.Quantity
+	for i := range pods {
+		pod := pods[i]
+		podQuantity, err := PodRequests(pod, resourceName)
+		if err != nil {
+			// log the error, but try to keep the most accurate count possible in log
+			// rationale here is that you may have had pods in a namespace that did not have
+			// explicit requests prior to adding the quota
+			glog.Infof("No explicit request for resource, pod %s/%s, %s", pod.Namespace, pod.Name, resourceName)
+		} else {
+			if sum == nil {
+				sum = podQuantity
+			} else {
+				err := sum.Add(*podQuantity)
+				if err != nil {
+					// we have exhausted more resource than could have been quota-ed, which should never happen
+					break
+				}
+			}
+		}
 	}
-	return resource.NewMilliQuantity(int64(val), resource.DecimalSI)
+	// if list is empty
+	if sum == nil {
+		q := resource.MustParse("0")
+		sum = &q
+	}
+	return sum
 }
 
-// IsPodCPUUnbounded returns true if the cpu use is unbounded for any container in pod
-func IsPodCPUUnbounded(pod *api.Pod) bool {
+// PodRequests returns sum of each resource request across all containers in pod
+func PodRequests(pod *api.Pod, resourceName api.ResourceName) (*resource.Quantity, error) {
+	if !PodHasRequests(pod, resourceName) {
+		return nil, fmt.Errorf("Each container in pod %s/%s does not have an explicit request for resource %s.", pod.Namespace, pod.Name, resourceName)
+	}
+	var sum *resource.Quantity
 	for j := range pod.Spec.Containers {
-		container := pod.Spec.Containers[j]
-		if container.Resources.Limits.Cpu().MilliValue() == int64(0) {
-			return true
+		value, _ := pod.Spec.Containers[j].Resources.Requests[resourceName]
+		if sum == nil {
+			sum = value.Copy()
+		} else {
+			err := sum.Add(value)
+			if err != nil {
+				return sum, err
+			}
 		}
 	}
-	return false
+	// if list is empty
+	if sum == nil {
+		q := resource.MustParse("0")
+		sum = &q
+	}
+	return sum, nil
 }
 
-// IsPodMemoryUnbounded returns true if the memory use is unbounded for any container in pod
-func IsPodMemoryUnbounded(pod *api.Pod) bool {
+// PodHasRequests verifies that each container in the pod has an explicit request that is non-zero for a named resource
+func PodHasRequests(pod *api.Pod, resourceName api.ResourceName) bool {
 	for j := range pod.Spec.Containers {
-		container := pod.Spec.Containers[j]
-		if container.Resources.Limits.Memory().Value() == int64(0) {
-			return true
+		value, valueSet := pod.Spec.Containers[j].Resources.Requests[resourceName]
+		if !valueSet {
```

Comment:
nit: the remainder of this function would be more readable as a single line:
return valueSet && value.Value() != int64(0)